test/e2e/conftest.py

```diff
@@ -30,24 +33,202 @@ def pytest_sessionstart(session):
     if os.getenv("CI"):
         ddtrace.patch_all()
 
+    if xdist.is_xdist_worker(session):
+        return
+
+    loop = asyncio.get_event_loop()
+    org_account: Account = loop.run_until_complete(create_organization_account())
+
+    _write_organization_account_to_test_session_config(org_account)
+
+
+def pytest_sessionfinish(session):
+    """
+    WARNING This hook runs BEFORE fixture teardown.
+
+    We use it to set a flag across processes that the global session has finished,
+    so we can safely cleanup shared setup. Otherwise this shared setup can be deleted
+    while some tests have not started yet due to parallelism.
+    """
```

Comment:
I don't quite follow what is the sequence here and what we are trying to solve by this file signaling.

Is `pytest_sessionfinish` called on the main process before all tests are finished?

Reply:
yes kinda, meaning fixture teardowns are executed after `pytest_sessionfinish` . i should improve the explanation here then

Reply:
found this https://github.com/pytest-dev/pytest/issues/7484, last commit uses it